BaseClasses.py

```diff
@@ -959,6 +959,7 @@ class Region:
     locations: List[Location]
     dungeon: Optional[Dungeon] = None
     shop: Optional = None
+    progress_type: LocationProgressType
```

Comment:
maybe `default_progress_type` since locations should still be able to override it?